lib/resque/worker.rb

```diff
@@ -141,7 +141,6 @@ def self.exists?(worker_id)
     # Workers should have `#prepare` called after they are initialized
     # if you are running work on the worker.
     def initialize(*queues)
-      initialize_queues_and_weights(queues)
```

Comment:
`queues` may be nil. IMHO you still need `@queues ||= []`

Reply:
@avgerinos i can add it but still upstream  resque does not have it Check https://github.com/resque/resque/blob/v1.27.4/lib/resque/worker.rb#L143-L157

Reply:
I would suggest to keep up with upstream's approach in case of non set queues that is to inform the user about the case. The respective message is the following:

`set QUEUE env var, e.g. $ QUEUE=critical,high rake resque:work`

Reply:
Also resque-pool https://github.com/skroutz/resque-pool/blob/master/lib/resque/pool.rb#L368 always passes the queues that reads from the config, to the worker correctly.

Reply:
Ignore this comment! Didn't see the setter method!